File: happy-emart/src/app/services/item.service.ts
import { Injectable } from '@angular/core';
import {HttpClient, HttpHeaders} from '@angular/common/http';
import {environment} from '../../environments/environment';

const httpOptions = {
    headers: new HttpHeaders({ 'Content-Type': 'application/json' })
  };
  
  @Injectable({
    providedIn: 'root'
  })
export class ItemService {

    constructor(private http: HttpClient) { }

    //buyer to search items
    searchItems(searchKey) {
        return this.http.get('${environment.baseUrl}/items/' + searchKey, httpOptions);
    }
   //buyer to view item Detail
    getItem(itemId: string):any{
        return this.http.get('${environment.baseUrl}/item/'+itemId);
      }

    //seller to add item
    addItem(item) {
      return this.http.post(`${environment.baseUrl}/items`, JSON.stringify(item), httpOptions);
    }

    //seller to update items
    updateItems(items) {
        return this.http.put(`${environment.baseUrl}/items`, JSON.stringify(items), httpOptions);
     }

    //seller to view all items
    viewItems() {
        return this.http.get(`${environment.baseUrl}/items`, httpOptions);
    }


}

File: emart-projects/Mid Tier/buyer-service/buyer-service/src/main/java/com/emart/buyer/service/ItemService.java
package com.emart.buyer.service;

import java.util.List;

import com.emart.buyer.entity.Item;


public interface ItemService {

	/**
	 * search Items
	 * @param String keyWords
	 */
    List<Item> searchItems(String keyWords);
    
	/**
	 * view Item Detail
	 * @param String keyWords
	 */
    Item viewItemDetail(Long itemId);

}
